asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py

```diff
@@ -0,0 +1,312 @@
+    plt.yscale("log")
+    plt.tight_layout()
+    path_to_plot = (
+        f"{MSE_FIGURES_PATH}/category_{category}_posts_per_user_data_collected.png"
+    )
+    plt.savefig(path_to_plot)
+    logger.info(
+        f"Plot with number of posts per user for category '{category}' saved successfully in {path_to_plot}"
+    )
+
+
+def plot_number_replies_per_user(mse_data: pd.DataFrame, category: str):
+    """
+    Plots and saves a histogram with number of replies per user.
+    The computation is based on data collected (i.e. replies collected from the specified category,
+    not the whole MSE)
+
+    Args:
+        mse_data (pd.DataFrame): MSE category/sub-forum
+    """
+    replies_per_user_data_collected = (
+        mse_data[mse_data["is_original_post"] == 0].groupby("user_id")[["id"]].count()
+    )
+
+    replies_per_user_data_collected.reset_index(inplace=True)
+    replies_per_user_data_collected.columns = ["user", "n_replies_user"]
+
+    plt.figure(figsize=(pc.figsize_x, pc.figsize_y))
+    plt.hist(
+        replies_per_user_data_collected["n_replies_user"],
+        bins=30,
+        color=pc.NESTA_COLOURS[0],
+        edgecolor="white",
+    )
+    plt.xlabel("Number of replies")
+    plt.ylabel("Number of users")
+    plt.title(f"Number of users with a given number of replies\nin {category} category")
+    plt.yscale("log")
+    plt.tight_layout()
+    path_to_plot = (
+        f"{MSE_FIGURES_PATH}/category_{category}_replies_per_user_data_collected.png"
+    )
+    plt.savefig(path_to_plot)
+    logger.info(
+        f"Plot with number of replies per user for category '{category}' saved successfully in {path_to_plot}"
+    )
+
+
+def create_argparser() -> argparse.ArgumentParser:
+    """
+    Creates an argument parser that can receive the following arguments:
+    - category: category or sub-forum (defaults to "green-ethical-moneysaving")
+    - collection_date_time: collection date/time (defaults to "2023_11_15")
+
+    Returns:
+        argparse.ArgumentParser: argument parser
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--category",
+        help="Category or sub-forum",
+        default="green-ethical-moneysaving",
+        type=str,
+    )
+
+    parser.add_argument(
+        "--collection_date_time",
+        help="Collection date/time",
+        default="2023_11_15",
+        type=str,
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+if __name__ == "__main__":
+    args = create_argparser()
+    category = args.category
+    collection_date_time = args.collection_date_time
+
+    MSE_FIGURES_PATH = f"{PROJECT_DIR}/outputs/figures/mse/{category}"
+    os.makedirs(MSE_FIGURES_PATH, exist_ok=True)
+
+    mse_data = get_mse_data(category, collection_date_time)
+    mse_data = add_datetime_variables(mse_data)
+
+    create_statistics_from_data(mse_data)
+
+    check_username_id_correspondence(mse_data)
+
+    plot_number_interactions_per_user(mse_data, category)
```

Comment:
Ok makes sense that I don't see that plot because it isn't used here! 
```suggestion
plot_number_posts_per_year(mse_data, category)   plot_number_interactions_per_user(mse_data, category)
```

Reply:
Good catch! :)